server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/IndexModeStatsActionType.java

```diff
@@ -0,0 +1,230 @@
+            final Map<IndexMode, IndexStats> stats = new EnumMap<>(IndexMode.class);
+            for (IndexMode mode : IndexMode.values()) {
+                stats.put(mode, new IndexStats());
+            }
+            for (NodeResponse node : getNodes()) {
+                for (Map.Entry<IndexMode, IndexStats> e : node.stats.entrySet()) {
+                    stats.get(e.getKey()).add(e.getValue());
+                }
+            }
+            return stats;
+        }
+    }
+
+    public static class NodeRequest extends TransportRequest {
+        NodeRequest() {
+
+        }
+
+        NodeRequest(StreamInput in) throws IOException {
+            super(in);
+        }
+    }
+
+    public static class NodeResponse extends BaseNodeResponse {
+        private final Map<IndexMode, IndexStats> stats;
+
+        NodeResponse(DiscoveryNode node, Map<IndexMode, IndexStats> stats) {
+            super(node);
+            this.stats = stats;
+        }
+
+        NodeResponse(StreamInput in, DiscoveryNode node) throws IOException {
+            super(in, node);
+            stats = in.readMap(IndexMode::readFrom, IndexStats::new);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeMap(stats, (o, m) -> IndexMode.writeTo(m, o), (o, s) -> s.writeTo(o));
+        }
+    }
+
+    public static class TransportAction extends TransportNodesAction<StatsRequest, StatsResponse, NodeRequest, NodeResponse, Void> {
+        private final ClusterService clusterService;
+        private final IndicesService indicesService;
+
+        @Inject
+        public TransportAction(
+            ClusterService clusterService,
+            TransportService transportService,
+            IndicesService indicesService,
+            ActionFilters actionFilters
+        ) {
+            super(
+                TYPE.name(),
+                clusterService,
+                transportService,
+                actionFilters,
+                NodeRequest::new,
+                transportService.getThreadPool().executor(ThreadPool.Names.MANAGEMENT)
+            );
+            this.clusterService = clusterService;
+            this.indicesService = indicesService;
+        }
+
+        @Override
+        protected StatsResponse newResponse(StatsRequest request, List<NodeResponse> nodeResponses, List<FailedNodeException> failures) {
+            return new StatsResponse(ClusterName.DEFAULT, nodeResponses, failures);
+        }
+
+        @Override
+        protected NodeRequest newNodeRequest(StatsRequest request) {
+            return new NodeRequest();
+        }
+
+        @Override
+        protected NodeResponse newNodeResponse(StreamInput in, DiscoveryNode node) throws IOException {
+            return new NodeResponse(in, node);
+        }
+
+        @Override
+        protected NodeResponse nodeOperation(NodeRequest request, Task task) {
```

Comment:
Do you think there is an easy way to reuse the logic in `IndicesStatsCache#internalGetIndicesStats()`? I think it is the same logic. 

I suspect we don't want the caching behaviour that `IndicesMetrics` has (because this should be invoked often), otherwise we could think of reusing it.

Reply:
++ I pushed https://github.com/elastic/elasticsearch/pull/116128/commits/650c7b74b0fba624c32aab2335228cc32fff7afa